### PR TITLE
Feature customer merge

### PR DESCRIPTION
**Pull request by khangaridb**

### CODE REVIEW

File: src/modules/customers/components/detail/CommonMerge.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  renderOptions(fieldName) {
+    const { datas } = this.props;
+    const options = [];
+
+    datas.forEach(data => {
+      options.push(<option key={data._id}>{data[fieldName] || ''}</option>);
+    });
+    options.push(
+      <option key="123" name="New">
```
Comment (ganzorig) [inline src/modules/customers/components/detail/CommonMerge.js]: Change key

File: src/modules/common/components/tagger/containers/Tagger.js [hunk 2]
```diff
@@ -65,7 +71,7 @@ export default compose(
     name: 'tagMutation',
     options: props => ({
       refetchQueries: [
-        props.type !== 'customer' ? `${props.type}` : `${props.type}s`
+        queries[props.type] ? queries[props.type] : `${props.type}`
```
Comment (batamar) [inline src/modules/common/components/tagger/containers/Tagger.js]: queries[props.type] || `${props.type}` ?

File: src/modules/companies/components/list/CompaniesList.js [hunk 3]
```diff
@@ -1,104 +1,238 @@
...
+  onChange() {
+    const { toggleAll, companies } = this.props;
+    toggleAll(companies, 'companies');
+  }
+
+  search(e) {
+    if (this.timer) clearTimeout(this.timer);
+    const { history } = this.props;
+    const searchValue = e.target.value;
+    this.setState({ searchValue });
```
Comment (batamar) [inline src/modules/companies/components/list/CompaniesList.js]: Add space above

File: src/modules/companies/components/list/CompaniesList.js [hunk 4]
```diff
@@ -1,104 +1,238 @@
...
+    this.removeCompanies = this.removeCompanies.bind(this);
+  }
+
+  onChange() {
+    const { toggleAll, companies } = this.props;
+    toggleAll(companies, 'companies');
+  }
+
+  search(e) {
+    if (this.timer) clearTimeout(this.timer);
```
Comment (batamar) [inline src/modules/companies/components/list/CompaniesList.js]: Add space below

File: src/modules/companies/components/list/CompaniesList.js [hunk 5]
```diff
@@ -1,104 +1,238 @@
...
+    }, 500);
+  }
+
+  removeCompanies(companies) {
+    const companyIds = [];
+
+    companies.forEach(company => {
+      companyIds.push(company._id);
+    });
+    this.props.removeCompanies({ companyIds });
```
Comment (batamar) [inline src/modules/companies/components/list/CompaniesList.js]: Add space above

**Pull request merged**